In [537]:
import pandas as pd

In [81]:
# Carregando dataset
#df_excel = pd.read_excel('HR_Employee_Data.xlsx')

# Transformando para .parquet
#df_excel.to_parquet("HR_Employee_Data.parquet")

#Carregando o dataset em .parquet
df = pd.read_parquet("HR_Employee_Data.parquet")

## **1. Verificando Colunas**

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14999 entries, 0 to 14998
Data columns (total 11 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Emp_Id                 14999 non-null  object 
 1   satisfaction_level     14999 non-null  float64
 2   last_evaluation        14999 non-null  float64
 3   number_project         14999 non-null  int64  
 4   average_montly_hours   14999 non-null  int64  
 5   time_spend_company     14999 non-null  int64  
 6   Work_accident          14999 non-null  int64  
 7   left                   14999 non-null  int64  
 8   promotion_last_5years  14999 non-null  int64  
 9   Department             14999 non-null  object 
 10  salary                 14999 non-null  object 
dtypes: float64(2), int64(6), object(3)
memory usage: 1.3+ MB


In [22]:
# Verificando valores nulos
df.isna().sum()

Emp_Id                   0
satisfaction_level       0
last_evaluation          0
number_project           0
average_montly_hours     0
time_spend_company       0
Work_accident            0
left                     0
promotion_last_5years    0
Department               0
salary                   0
dtype: int64

In [20]:
df.dtypes

Emp_Id                   category
satisfaction_level        float64
last_evaluation           float64
number_project              int64
average_montly_hours        int64
time_spend_company          int64
Work_accident               int64
left                        int64
promotion_last_5years       int64
Department               category
salary                   category
dtype: object

In [82]:
# Mudando as colunas para categoria a fim de otimizar o consumo de memória
df['Emp_Id'] = df['Emp_Id'].astype('category')
df['Department'] = df['Department'].astype('category')
df['salary'] = df['salary'].astype('category')

In [83]:
#Criando a coluna de diferença entre a avaliação mais recente e a anterior
df['diff_evaluation'] = df['satisfaction_level'] - df['last_evaluation']
nova_ordem = ['Emp_Id','satisfaction_level','last_evaluation','diff_evaluation','number_project','average_montly_hours','time_spend_company', 'Work_accident','left', 'promotion_last_5years','Department','salary']
df = df[nova_ordem]

Emp_Id  satisfaction_level  last_evaluation  diff_evaluation  \
0      IND02438                0.38             0.53            -0.15   
1      IND28133                0.80             0.86            -0.06   
2      IND07164                0.11             0.88            -0.77   
3      IND30478                0.72             0.87            -0.15   
4      IND24003                0.37             0.52            -0.15   
...         ...                 ...              ...              ...   
14994  IND40221                0.40             0.57            -0.17   
14995  IND24196                0.37             0.48            -0.11   
14996  IND33544                0.37             0.53            -0.16   
14997  IND40533                0.11             0.96            -0.85   
14998  IND11649                0.37             0.52            -0.15   

       number_project  average_montly_hours  time_spend_company  \
0                   2                   157                   3   
1                   5                   262                   6   
2                   7                   272                   4   
3                   5                   223                   5   
4                   2                   159                   3   
...               ...                   ...                 ...   
14994               2                   151                   3   
14995               2                   160                   3   
14996               2                   143                   3   
14997               6                   280                   4   
14998               2                   158                   3   

       Work_accident  left  promotion_last_5years Department  salary  
0                  0     1                      0      sales     low  
1                  0     1                      0      sales  medium  
2                  0     1                      0      sales  medium  
3                  0     1                      0      sales     low  
4                  0     1                      0      sales     low  
...              ...   ...                    ...        ...     ...  
14994              0     1                      0    support     low  
14995              0     1                      0    support     low  
14996              0     1                      0    support     low  
14997              0     1                      0    support     low  
14998              0     1                      0    support     low  

[14999 rows x 12 columns]

## **2. Perguntas de Negócio**

### **2.1 Geral**

1. Qual o total de demissões?
2. Quantos departamentos existem?
3. Qual a média do nível de satisfação?
4. Qual a média de hora trabalhada por mês?
5. Na media, a tendência é de que os trabalhadores fiquem mais satisfeitos ou mais infelizes?
6. Qual o maior quantidade de tempo trabalhado em um mês?
7. Qual a média de anos que um trabalhador fica na empresa?
8. A maior parte dos empregados possuem quanto tempo de empresa?
9. Qual o total de acidentes de trabalho já ocorridos?
10. Quantas promoções tiveram no total?

### **2.2 Por Setor** 

1. Qual setor possui mais demissões?
2. Qual setor com mais pessoas?
3. Em qual setor está registrada a melhor média de satisfação? E a pior?
4. Em qual setor as pessoas mais melhoraram suas avaliações em relação as anteriores? E o pior?
5. Em média, qual setor acumula mais horas trabalhadas mensalmente?
6. Em média, qual setor possui os trabalhadores mais longevos? E qual o setor mais rotativo?
7. Qual o setor mais perigoso da empresa?
8. Qual setor possui a maior taxa de promoções?

### **2.3 Por Salário**

1. Qual percentual de colaboradores por salário?
2. Em qual faixa salarial há mais demissões?
3. Em qual salário está registrada a melhor média de satisfação? E a pior?
4. Em qual remuneração as pessoas mais melhoraram suas avaliações em relação as anteriores? E o pior?
5. Em média, categoria salarial acumula mais horas trabalhadas mensalmente?
6. Em média, qual salário possui os trabalhadores mais longevos? E qual o faixa salarial mais rotativo?
7. Qual o faixa salarial mais perigoso da empresa?
8. Qual remuneração possui a maior taxa de promoções?

## **3. Respondendo as Perguntas**

### **3.1 Geral**
1. Qual o total de demissões?
2. Quantos departamentos existem?
3. Qual a média do nível de satisfação?
4. Qual a média de hora trabalhada por mês?
5. Na media, a tendência é de que os trabalhadores fiquem mais satisfeitos ou mais infelizes?
6. Qual o maior quantidade de tempo trabalhado em um mês?
7. Qual a média de anos que um trabalhador fica na empresa?
8. A maior parte dos empregados possuem quanto tempo de empresa?
9. Qual o total de acidentes de trabalho já ocorridos?
10. Quantas promoções tiveram no total?

In [233]:
# 1. Qual o total de demissões?
print(f'Houveram {df["left"].value_counts()[1]} demissões ao todo.')

Houveram 3571 demissões ao todo.


In [234]:
# 2. Quantos departamentos existem?
print(f'Existem {df["Department"].nunique()} departamentos ao todo.')

Existem 10 departamentos ao todo.


In [235]:
# 3. Qual a média do nível de satisfação?
print(f'A média do nível de satisfação é de {df["satisfaction_level"].mean():,.2f}%.')

A média do nível de satisfação é de 0.61%.


In [236]:
# 4. Qual a média de hora trabalhada por mês?
print(f'A média hora trabalhada por mês é de {df["average_montly_hours"].mean()/21:,.2f} horas por dia.')

A média hora trabalhada por mês é de 9.57 horas por dia.


In [237]:
# 5. Na media, a tendência é de que os trabalhadores fiquem mais satisfeitos ou mais descontentes?
print(f'A média está na casa de {df["diff_evaluation"].mean():,.2f}; \n'
      'Portanto, a tendência é de que as pessoas ficaram mais descontentes entre uma avaliação e outra.')

A média está na casa de -0.10; 
Portanto, a tendência é de que as pessoas ficaram mais descontentes entre uma avaliação e outra.


In [238]:
# 6. Qual o maior quantidade de tempo trabalhado em um mês?
print(f'A maior média trabalhada em um único mês é de {df["average_montly_hours"].max()} horas; \n'
      f'O que dá certa de {df["average_montly_hours"].max()/21:,.2f} horas por dia')

A maior média trabalhada em um único mês é de 310 horas; 
O que dá certa de 14.76 horas por dia


In [240]:
# 7. Qual a média de anos que um trabalhador fica na empresa?
print(f'A permanência média de um colaborador é de {df.loc[df["left"]==1, "time_spend_company"].mean():,.1f} anos.')

A permanência média de um colaborador é de 3.9 anos.


In [241]:
# 8. A maior parte dos empregados possuem quanto tempo de empresa?
print(f'A maior parte dos empregados possui {df["time_spend_company"].mode()[0]} anos de empresa.')

A maior parte dos empregados possui 3 anos de empresa.


In [242]:
# 9. Qual o total de acidentes de trabalho já ocorridos?
print(f'Ao todo, houveram {df["Work_accident"].value_counts()[1]} acidentes na empresa.')

Ao todo, houveram 2169 acidentes na empresa.


In [243]:
# 10. Quantas promoções tiveram no total?
print(f'Houveram {df["promotion_last_5years"].value_counts()[1]} promoções nos últimos 5 anos.')

Houveram 319 promoções nos últimos 5 anos.


### **3.2 Por Setor**
1. Qual setor possui mais demissões?
2. Qual setor com mais pessoas?
3. Em qual setor está registrada a melhor média de satisfação? E a pior?
4. Em qual setor há mais pessoas com 100% de satisfação?
5. Em qual setor as pessoas mais melhoraram suas avaliações em relação as anteriores? E o pior?
6. Em média, qual setor acumula mais horas trabalhadas mensalmente?
7. Em média, qual setor possui os trabalhadores mais longevos? E qual o setor mais rotativo?
8. Qual o setor mais perigoso da empresa?
9. Qual setor possui a maior taxa de promoções?

In [290]:
# 1. Qual setor possui mais demissões?
df1 = (df.loc[df['left']==1, ['Department','left']].groupby('Department')
                                                   .count().sort_values(by='left',ascending=False)
                                                   .reset_index())
print(f'O setor que mais houve demissões foi o setor de {df1["Department"][0]}, com {df1["left"][0]} demissões.\n'
      f'Isso representa {(df1["left"][0]/df1["left"].sum())*100:,.2f}% do total de demissões.')

O setor que mais houve demissões foi o setor de sales, com 1014 demissões.
Isso representa 28.40% do total de demissões.


In [179]:
# 2. Qual setor com mais pessoas?
df1 = df.loc[:, ['Department','Emp_Id']].groupby('Department').count().sort_values(by='Emp_Id',ascending=False).reset_index()
print(f'O setor com mais pessoas é o de {df1["Department"][0]}, com {df1["Emp_Id"][0]} colaboradores. \n'
      f'Isso corresponde a {(df1["Emp_Id"][0]/df["Emp_Id"].nunique()):,.2f}% do total de colaboradores.')

O setor com mais pessoas é o de sales, com 4140 colaboradores. 
Isso corresponde a 0.28% do total de colaboradores.


In [302]:
# 3. Em qual setor está registrada a melhor média de satisfação? E a pior?
df1 = (df.loc[:, ['Department','satisfaction_level']].groupby('Department').mean().sort_values(by='satisfaction_level',ascending=False).reset_index())
print(f'O setor com melhor média de satisfação é o de {df1["Department"][0]}, com {(df1["satisfaction_level"][0])*100:,.2f}% de satisfação dos colaboradores. \n'
      f'Já a pior média, fica com {df1["Department"].iloc[-1]}, com {(df1["satisfaction_level"].min())*100:,.2f}% de satisfação.')

O setor com melhor média de satisfação é o de management, com 62.13% de satisfação dos colaboradores. 
Já a pior média, fica com accounting, com 58.22% de satisfação.


In [288]:
# 4. Em qual setor há mais pessoas com 100% de satisfação?
df1 = (df.loc[df['satisfaction_level']==1, ['Department','satisfaction_level']].groupby('Department')
                                                                               .count()
                                                                               .sort_values(by='satisfaction_level',ascending=False)
                                                                               .reset_index())
print(f'O setor com mais colaboradoressatisfeitos é o de {df1["Department"][0]},' 
      f' com {(df1["satisfaction_level"][0])} notas máximas de satisfação dos colaboradores. \n'
      f'Já o pior, fica com {df1["Department"].iloc[-1]}, com {(df1["satisfaction_level"].iloc[-1])} notas máximas.')

O setor com mais colaboradoressatisfeitos é o de sales, com 30 notas máximas de satisfação dos colaboradores. 
Já o pior, fica com accounting, com 3 notas máximas.


In [289]:
# 4. Em qual setor as pessoas mais melhoraram suas avaliações em relação as anteriores? E o pior?
df1 = (df.loc[:, ['Department','diff_evaluation']].groupby('Department').mean()
                                                                        .sort_values(by='diff_evaluation',ascending=False)
                                                                        .reset_index())
print(f'O setor em que as avaliações de satisfação tiveram uma melhora em relação a anterior é o de {df1["Department"][0]},'
      f' com {(df1["diff_evaluation"][0])*100:,.2f}% de satisfação dos colaboradores. \n'
      f'Já o que mais diminuiu, foi o de {df1["Department"].iloc[-1]}, com {(df1["diff_evaluation"].min())*100:,.2f}% de satisfação.')

O setor em que as avaliações de satisfação tiveram uma melhora em relação a anterior é o de RandD, com -9.23% de satisfação dos colaboradores. 
Já o que mais diminuiu, foi o de accounting, com -13.56% de satisfação.


In [207]:
# 5. Em média, qual setor acumula mais horas trabalhadas mensalmente?
df1 = df.loc[:, ['Department','average_montly_hours']].groupby('Department').mean().sort_values(by='average_montly_hours',ascending=False).reset_index()
print(f'Na média, o setor em que acumulou mais horas trabalhadas mensalmente é o de {df1["Department"][0]}, com {(df1["average_montly_hours"][0]):,.2f} horas. \n'
      f'Já o que mais diminuiu, foi o de {df1["Department"].iloc[-1]}, com {(df1["average_montly_hours"].iloc[-1]):,.2f} de horas trabalhadas.')

Na média, o setor em que acumulou mais horas trabalhadas mensalmente é o de technical, com 202.50 horas. 
Já o que mais diminuiu, foi o de hr, com 198.68 de horas trabalhadas.


In [430]:
# 6. Em média, qual setor possui os trabalhadores mais longevos? E qual o setor mais rotativo?
df1 = df.loc[df['left']==1, ['Department','time_spend_company']].groupby('Department').mean().sort_values(by='time_spend_company',ascending=False).reset_index()
print(f'Na média, o setor em que se tem as pessoas mais longevas é o de {df1["Department"][0]}, com {(df1["time_spend_company"][0]):,.1f} anos. \n'
      f'Já o que mais rotativo, é o de {df1["Department"].iloc[-1]}, com {(df1["time_spend_company"].iloc[-1]):,.1f} de anos de empresa.')

Na média, o setor em que se tem as pessoas mais longevas é o de RandD, com 4.0 anos. 
Já o que mais rotativo, é o de hr, com 3.8 de anos de empresa.


In [214]:
# 7. Qual o setor mais perigoso da empresa?
df1 = df.loc[:, ['Department','Work_accident']].groupby('Department').count().sort_values(by='Work_accident',ascending=False).reset_index()
print(f'O setor em que se tem mais acidentes é o de {df1["Department"][0]}, com {(df1["Work_accident"][0])} acidentes registrados. \n'
      f'Já o que mais seguro, é o de {df1["Department"].iloc[-1]}, com {(df1["Work_accident"].iloc[-1])} acidentes registrados.')

O setor em que se tem mais acidentes é o de sales, com 4140 acidentes registrados. 
Já o que mais seguro, é o de management, com 630 acidentes registrados.


In [274]:
# 8. Qual setor possui a maior taxa de promoções?
df1 = df.loc[df['promotion_last_5years']==1, ['Department','promotion_last_5years']].groupby('Department').count().sort_values(by='promotion_last_5years',ascending=False).reset_index()
print(f'O setor em que há mais promoções é o de {df1["Department"][0]}, com {(df1["promotion_last_5years"][0]/df1["promotion_last_5years"].sum())*100:,.2f}% das promoções. \n'
      f'Já o que menos promove, é o de {df1["Department"].iloc[-1]}, com {(df1["promotion_last_5years"].iloc[-1]/df1["promotion_last_5years"].sum())*100:,.2f}% das promoções.')

O setor em que há mais promoções é o de sales, com 31.35% das promoções. 
Já o que menos promove, é o de product_mng, com 0.00% das promoções.


### **3.3 Por Salário**
1. Qual percentual de colaboradores por salário?
2. Em qual faixa salarial há mais demissões?
3. Em qual salário está registrada a melhor média de satisfação? E a pior?
4. Em qual remuneração as pessoas mais melhoraram suas avaliações em relação as anteriores? E o pior?
5. Em média, categoria salarial acumula mais horas trabalhadas mensalmente?
6. Em média, qual salário possui os trabalhadores mais longevos? E qual o faixa salarial mais rotativo?
7. Qual o faixa salarial mais perigoso da empresa?
8. Qual remuneração possui a maior taxa de promoções?

In [255]:
# 1. Qual percentual de colaboradores por salário?
df2 = df.loc[:, ['salary','Emp_Id']].groupby('salary').count().sort_values(by='Emp_Id',ascending=False).reset_index()
print(f'O setor que mais houve demissões foi o {df2["salary"][0]}, com {df2["Emp_Id"][0]} colaboradores.\n'
      f'Isso representa {(df2["Emp_Id"][0]/df2["Emp_Id"].sum())*100:,.2f}% do total.')

O setor que mais houve demissões foi o low, com 7316 demissões.
Isso representa 48.78% do total de demissões.


In [261]:
# 2. Em qual faixa salarial há mais demissões?
df2 = df.loc[df['left']==1, ['salary','left']].groupby('salary').count().sort_values(by='left',ascending=False).reset_index()
print(f'A faixa salarial em que mais houveram mais demissões foi o {df2["salary"][0]}, com {df2["left"][0]} demissões.\n'
      f'Isso representa {(df2["left"][0]/df2["left"].sum())*100:,.2f}% do total de demissões.')

A faixa salarial em que mais houveram mais demissões foi o low, com 2172 demissões.
Isso representa 60.82% do total de demissões.


In [258]:
# 3. Em qual salário está registrada a melhor média de satisfação? E a pior?
df2 = df.loc[:, ['salary','satisfaction_level']].groupby('salary').mean().sort_values(by='satisfaction_level',ascending=False).reset_index()
print(f'O salário com melhor média de satisfação é o de {df2["salary"][0]}, com {(df2["satisfaction_level"][0])*100:,.2f}% de satisfação dos colaboradores. \n'
      f'Já a pior média, fica com {df2["salary"].iloc[-1]}, com {(df2["satisfaction_level"].min())*100:,.2f}% de satisfação.')

O salário com melhor média de satisfação é o de high, com 63.75% de satisfação dos colaboradores. 
Já a pior média, fica com low, com 60.08% de satisfação.


In [278]:
# 4. Em qual remuneração as pessoas mais melhoraram suas avaliações em relação as anteriores? E o pior?
df2 = df.loc[:, ['salary','diff_evaluation']].groupby('salary').mean().sort_values(by='diff_evaluation',ascending=False).reset_index()
print(f'O setor em que as avaliações de satisfação tiveram uma melhora em relação a anterior é o de' 
      f'{df2["salary"][0]}, com {(df2["diff_evaluation"][0])*100:,.2f}% de satisfação dos colaboradores. \n'
      f'Já o que mais diminuiu, foi o de {df2["salary"].iloc[-1]}, com {(df2["diff_evaluation"].min())*100:,.2f}% de satisfação.')

O setor em que as avaliações de satisfação tiveram uma melhora em relação a anterior é o dehigh, com -6.69% de satisfação dos colaboradores. 
Já o que mais diminuiu, foi o de low, com -11.63% de satisfação.


In [268]:
# 5. Em média, qual categoria salarial acumula mais horas trabalhadas mensalmente?
df2 = df.loc[:, ['salary','average_montly_hours']].groupby('salary').mean().sort_values(by='average_montly_hours',ascending=False).reset_index()
print(f'Na média, a remuneração em que acumulou mais horas trabalhadas mensalmente é a {df2["salary"][0]}, com {(df2["average_montly_hours"][0]):,.2f} horas. \n'
      f'Já o que mais diminuiu, foi o de {df2["salary"].iloc[-1]}, com {(df2["average_montly_hours"].iloc[-1]):,.2f} de horas trabalhadas.')

Na média, a remuneração em que acumulou mais horas trabalhadas mensalmente é a medium, com 201.34 horas. 
Já o que mais diminuiu, foi o de high, com 199.87 de horas trabalhadas.


In [267]:
# 6. Em média, qual salário possui os trabalhadores mais longevos? E qual o faixa salarial mais rotativo?
df2 = df.loc[:, ['salary','time_spend_company']].groupby('salary').mean().sort_values(by='time_spend_company',ascending=False).reset_index()
print(f'Na média, a faixa salarial mais longeva é a {df2["salary"][0]}, com {(df2["time_spend_company"][0]):,.1f} anos. \n'
      f'Já a mais rotativo, é o {df2["salary"].iloc[-1]}, com {(df2["time_spend_company"].iloc[-1]):,.1f} de anos de empresa.')

Na média, a faixa salarial mais longeva é a high, com 3.7 anos. 
Já a mais rotativo, é o low, com 3.4 de anos de empresa.


In [271]:
# 7. Qual o faixa salarial mais perigoso da empresa?
df2 = df.loc[:, ['salary','Work_accident']].groupby('salary').count().sort_values(by='Work_accident',ascending=False).reset_index()
print(f'O salário em que houveram mais acidentes é o {df2["salary"][0]}, com {(df2["Work_accident"][0])} acidentes registrados. \n'
      f'Já o mais seguro, é o {df2["salary"].iloc[-1]}, com {(df2["Work_accident"].iloc[-1])} acidentes registrados.')

O salário em que houveram mais acidentes é o low, com 7316 acidentes registrados. 
Já o mais seguro, é o high, com 1237 acidentes registrados.


In [273]:
# 8. Qual remuneração possui a maior taxa de promoções?
df2 = df.loc[df['promotion_last_5years']==1, ['salary','promotion_last_5years']].groupby('salary').count().sort_values(by='promotion_last_5years',ascending=False).reset_index()
print(f'A faixa em que se tem mais promoções é a {df2["salary"][0]}, com {(df2["promotion_last_5years"][0]/df2["promotion_last_5years"].sum())*100:,.2f}% das promoções. \n'
      f'Já a que menos promove, é a {df2["salary"].iloc[-1]}, com {(df2["promotion_last_5years"].iloc[-1]/df2["promotion_last_5years"].sum())*100:,.2f}% das promoções.')

A faixa em que se tem mais promoções é a medium, com 56.74% das promoções. 
Já a que menos promove, é a low, com 20.69% das promoções.


## **4. Estatística Descritiva**

In [422]:
stats = df.describe().T
stats['coef_var'] = stats['std'] / stats['mean']
stats

count        mean        std    min     25%     50%  \
satisfaction_level     14999.0    0.612834   0.248631   0.09    0.44    0.64   
last_evaluation        14999.0    0.716102   0.171169   0.36    0.56    0.72   
diff_evaluation        14999.0   -0.103268   0.286665  -0.89   -0.23   -0.08   
number_project         14999.0    3.803054   1.232592   2.00    3.00    4.00   
average_montly_hours   14999.0  201.050337  49.943099  96.00  156.00  200.00   
time_spend_company     14999.0    3.498233   1.460136   2.00    3.00    3.00   
Work_accident          14999.0    0.144610   0.351719   0.00    0.00    0.00   
left                   14999.0    0.238083   0.425924   0.00    0.00    0.00   
promotion_last_5years  14999.0    0.021268   0.144281   0.00    0.00    0.00   

                          75%     max  coef_var  
satisfaction_level       0.82    1.00  0.405707  
last_evaluation          0.87    1.00  0.239029  
diff_evaluation          0.08    0.62 -2.775931  
number_project           5.00    7.00  0.324106  
average_montly_hours   245.00  310.00  0.248411  
time_spend_company       4.00   10.00  0.417392  
Work_accident            0.00    1.00  2.432193  
left                     0.00    1.00  1.788977  
promotion_last_5years    0.00    1.00  6.783943

In [420]:
# Avaliando a coluna de 'satisfaction_level'
df_satisf = (df.loc[:, ['Department','satisfaction_level']].groupby(['Department'])
                                                     .agg({'satisfaction_level':['mean','median','std','count']})
                                                     .reset_index())
df_satisf['satisfaction_level',   'coef_var'] = df_satisf['satisfaction_level',   'std'] / df_satisf['satisfaction_level',   'mean']
df_satisf

Department satisfaction_level                                 
                             mean median       std count  coef_var
0           IT           0.618142  0.660  0.249903  1227  0.404281
1        RandD           0.619822  0.650  0.245269   787  0.395708
2   accounting           0.582151  0.610  0.255235   767  0.438435
3           hr           0.598809  0.610  0.247929   739  0.414037
4   management           0.621349  0.655  0.227747   630  0.366536
5    marketing           0.618601  0.640  0.244280   858  0.394891
6  product_mng           0.619634  0.640  0.242279   902  0.391003
7        sales           0.614447  0.640  0.250251  4140  0.407278
8      support           0.618300  0.650  0.246406  2229  0.398522
9    technical           0.607897  0.640  0.254193  2720  0.418152

In [421]:
# Avaliando a coluna de Satisfação
df_last = (df.loc[:, ['Department','last_evaluation']].groupby(['Department'])
                                                     .agg({'last_evaluation':['mean','median','std','count']})
                                                     .reset_index())
df_last['last_evaluation',   'coef_var'] = df_last['last_evaluation',   'std'] / df_last['last_evaluation',   'mean']
df_last

Department last_evaluation                                 
                          mean median       std count  coef_var
0           IT        0.716830   0.72  0.164508  1227  0.229494
1        RandD        0.712122   0.71  0.165093   787  0.231832
2   accounting        0.717718   0.73  0.171904   767  0.239515
3           hr        0.708850   0.72  0.174584   739  0.246292
4   management        0.724000   0.73  0.160263   630  0.221357
5    marketing        0.715886   0.71  0.173447   858  0.242284
6  product_mng        0.714756   0.72  0.178147   902  0.249242
7        sales        0.709717   0.70  0.171459  4140  0.241588
8      support        0.723109   0.74  0.171224  2229  0.236789
9    technical        0.721099   0.73  0.173364  2720  0.240417

In [375]:
# Análise da coluna 'average_montly_hours' por setor
df_month = df.loc[:, ['Department','average_montly_hours']].groupby(['Department']).describe().reset_index()
df_month['average_montly_hours',   'coef_var'] = df_month['average_montly_hours',   'std'] / df_month['average_montly_hours',   'mean']
df_month

Department average_montly_hours                                       \
                              count        mean        std   min     25%   
0           IT               1227.0  202.215974  50.688311  96.0  160.00   
1        RandD                787.0  200.800508  49.253223  98.0  157.00   
2   accounting                767.0  201.162973  51.108274  97.0  153.50   
3           hr                739.0  198.684709  50.366214  98.0  152.00   
4   management                630.0  201.249206  47.382452  97.0  161.00   
5    marketing                858.0  199.385781  49.356930  96.0  154.00   
6  product_mng                902.0  199.965632  50.112636  98.0  155.00   
7        sales               4140.0  200.911353  49.560393  96.0  156.00   
8      support               2229.0  200.758188  50.023687  96.0  155.00   
9    technical               2720.0  202.497426  50.596130  97.0  157.75   

                                   
     50%     75%    max  coef_var  
0  199.0  245.00  308.0  0.250664  
1  200.0  248.00  308.0  0.245284  
2  199.0  247.00  310.0  0.254064  
3  197.0  242.00  310.0  0.253498  
4  204.0  243.00  307.0  0.235442  
5  198.0  242.00  310.0  0.247545  
6  198.0  244.00  310.0  0.250606  
7  201.0  245.00  310.0  0.246678  
8  200.0  246.00  310.0  0.249174  
9  201.0  246.25  310.0  0.249861

In [442]:
# Análise da coluna 'diff_evaluation' por setor
df_dif = df.loc[:, ['Department','diff_evaluation']].groupby(['Department']).describe().reset_index()
df_dif['diff_evaluation',   'coef_var'] = df_dif['diff_evaluation',   'std'] / df_dif['diff_evaluation',   'mean']
df_dif

Department diff_evaluation                                                \
                         count      mean       std   min   25%   50%     75%   
0           IT          1227.0 -0.098688  0.293305 -0.87 -0.23 -0.08  0.0900   
1        RandD           787.0 -0.092300  0.286502 -0.88 -0.25 -0.06  0.1000   
2   accounting           767.0 -0.135567  0.281147 -0.87 -0.27 -0.10  0.0300   
3           hr           739.0 -0.110041  0.278295 -0.87 -0.22 -0.10  0.0700   
4   management           630.0 -0.102651  0.282688 -0.87 -0.25 -0.07  0.0800   
5    marketing           858.0 -0.097284  0.274242 -0.87 -0.22 -0.08  0.0700   
6  product_mng           902.0 -0.095122  0.274238 -0.87 -0.23 -0.08  0.0775   
7        sales          4140.0 -0.095271  0.288067 -0.89 -0.22 -0.07  0.0900   
8      support          2229.0 -0.104809  0.283181 -0.88 -0.22 -0.08  0.0700   
9    technical          2720.0 -0.113202  0.296235 -0.88 -0.25 -0.08  0.0700   

                   
    max  coef_var  
0  0.51 -2.972043  
1  0.59 -3.104030  
2  0.60 -2.073856  
3  0.53 -2.529017  
4  0.58 -2.753878  
5  0.50 -2.818971  
6  0.62 -2.883013  
7  0.62 -3.023673  
8  0.55 -2.701868  
9  0.59 -2.616866

In [443]:
# Análise da coluna 'time_spend_company' por setor
df_time = df.loc[df['left']==1, ['Department','time_spend_company']].groupby(['Department']).describe().reset_index()
df_time['time_spend_company','coef_var'] = df_time['time_spend_company','std'] / df_time['time_spend_company','mean']
display(df_time)

Department time_spend_company                                          \
                            count      mean       std  min  25%  50%  75%   
0           IT              273.0  3.860806  0.908924  2.0  3.0  4.0  5.0   
1        RandD              121.0  4.000000  0.966092  2.0  3.0  4.0  5.0   
2   accounting              204.0  3.794118  0.907964  2.0  3.0  4.0  4.0   
3           hr              215.0  3.753488  0.942097  3.0  3.0  3.0  4.0   
4   management               91.0  3.758242  0.834541  2.0  3.0  4.0  4.0   
5    marketing              203.0  3.857143  1.031334  3.0  3.0  3.0  5.0   
6  product_mng              198.0  3.989899  1.027490  3.0  3.0  4.0  5.0   
7        sales             1014.0  3.812623  0.978747  2.0  3.0  4.0  5.0   
8      support              555.0  3.933333  1.046339  2.0  3.0  4.0  5.0   
9    technical              697.0  3.959828  0.955080  2.0  3.0  4.0  5.0   

                  
   max  coef_var  
0  6.0  0.235423  
1  6.0  0.241523  
2  6.0  0.239308  
3  6.0  0.250992  
4  6.0  0.222056  
5  6.0  0.267383  
6  6.0  0.257523  
7  6.0  0.256712  
8  6.0  0.266018  
9  6.0  0.241192

In [444]:
# Análise da coluna 'Work_accident' por setor
df_accident = (df.loc[df['Work_accident']==1, ['Department','Work_accident']].groupby(['Department'])
                                                                             .count()
                                                                             .sort_values(by='Work_accident', ascending=True)
                                                                             .reset_index())
df_accident['percent_colaboradores(%)'] = round(((df_accident['Work_accident'] / df['Emp_Id'].count())*100),2)
df_accident['percent_total(%)'] = round(((df_accident['Work_accident'] / df_accident['Work_accident'].sum())*100),2)
df_accident

Department  Work_accident  percent_colaboradores(%)  percent_total(%)
0           hr             89                      0.59              4.10
1   accounting             96                      0.64              4.43
2   management            103                      0.69              4.75
3  product_mng            132                      0.88              6.09
4        RandD            134                      0.89              6.18
5    marketing            138                      0.92              6.36
6           IT            164                      1.09              7.56
7      support            345                      2.30             15.91
8    technical            381                      2.54             17.57
9        sales            587                      3.91             27.06

In [450]:
df_promo = (df.loc[df['promotion_last_5years']==1, ['Department','promotion_last_5years']].groupby(['Department'])
                                                                             .count()
                                                                             .sort_values(by='promotion_last_5years', ascending=True)
                                                                             .reset_index())
df_promo['percent_colaboradores(%)'] = round(((df_promo['promotion_last_5years'] / df['Emp_Id'].count())*100),2)
df_promo['percent_total(%)'] = round(((df_promo['promotion_last_5years'] / df_promo['promotion_last_5years'].sum())*100),2)
df_promo

Department  promotion_last_5years  percent_colaboradores(%)  \
0  product_mng                      0                      0.00   
1           IT                      3                      0.02   
2   accounting                     14                      0.09   
3           hr                     15                      0.10   
4      support                     20                      0.13   
5        RandD                     27                      0.18   
6    technical                     28                      0.19   
7    marketing                     43                      0.29   
8   management                     69                      0.46   
9        sales                    100                      0.67   

   percent_total(%)  
0              0.00  
1              0.94  
2              4.39  
3              4.70  
4              6.27  
5              8.46  
6              8.78  
7             13.48  
8             21.63  
9             31.35

In [514]:
df.sample()

Emp_Id  satisfaction_level  last_evaluation  diff_evaluation  \
1653  IND33228                 0.1             0.88            -0.78   

      number_project  average_montly_hours  time_spend_company  Work_accident  \
1653               7                   282                   4              0   

      left  promotion_last_5years Department salary  
1653     1                      0    support    low

In [536]:
df_perfil = (df.loc[df['left']==1, ['satisfaction_level','last_evaluation','diff_evaluation',
                                    'number_project','average_montly_hours','time_spend_company',
                                    'Department','Emp_Id',]].groupby(['Department'])
                                                  .agg({'Emp_Id':['count'],
                                                        'satisfaction_level':['mean'],
                                                        'last_evaluation':['mean'],
                                                        'diff_evaluation':['mean'],
                                                        'number_project':['mean'],
                                                        'average_montly_hours':['mean'],
                                                        'time_spend_company':['mean']})
                                                  .reset_index())
print(f'Média de Satisfação: {round((df_perfil["satisfaction_level"].mean())*100,2)[0]}%')
print(f'Média da Satisfação Anterior: {round((df_perfil["last_evaluation"].mean()*100),2)[0]}%')
print(f'Média de Diferença: {round((df_perfil["diff_evaluation"].mean())*100,2)[0]}%')
print(f'Média de participação em projetos: {round((df_perfil["number_project"].mean()),2)[0]}')
print(f'Média de horas mensais: {round((df_perfil["average_montly_hours"].mean()),2)[0]} horas mensais.')
print(f'Média de tempo na empresa: {round((df_perfil["time_spend_company"].mean()),2)[0]} anos')

Média de Satisfação: 43.69%
Média da Satisfação Anterior: 71.68%
Média de Diferença: -27.99%
Média de participação em projetos: 3.87
Média de horas mensais: 207.01 horas mensais.
Média de tempo na empresa: 3.87 anos


## **5. Conclusão**
##### **1.** O nível de satisfação como um todo caiu quase 11% desde a última avaliação e não apenas isso, as respostas estão menos homogêneas, ou seja, há menos consenso e unidade nessa categoria, com uma amplitude maior nas respostas. Isso se dá, não apenas pelo aumento no desvio padrão e o coeficiente de variação, como também se observarmos que na última avaliação, a resposta mais baixa foi 36% de satisfação, agora, a mínima é de 9%, um aumento de 4x na amplitude total das respostas.<br>
##### **2.** Todos os setores apresentaram queda na satisfação. O setor de **Accounting** foi o que mais caiu, com média de 13% de queda em relação a avaliação anterior, com respostas chegando a 87% abaixo da anterior. O que menos teve alteração foi o de **Product_mng**, com redução de apenas 9%, mas que também alcançou picos de 87% de diferença.<br>
##### **3.** Os colaboradores do setor de **Management** não apenas são os que, em média, trabalham mais horas, como seus resultados são os mais homogêneos, ou seja, a média de cada trabalhador próxima da média do setor. Além disso, é o setor com menos pessoal, o que o coloca como o setor com os dados mais consistentes nesta categoria.
##### **4.** Das pessoas que se desligaram, a média de permanência até o desligamento vai de 3.8 a 4 anos, com tempo mínimo de 2 anos e máximo de 6. Sendo o **RH** o setor mais rotativo e o **RandD** o mai longevo.
##### **5.** **Sales**, **Technical** e **Supprt** possuem as maiores taxas de acidentes entre os setores. Juntos, somam quase 60% do total de acidentes. Sales é o mais perigoso, de cada 100 colaboradores, três sofreram algum tipo de acidente de trabalho.
##### **6.** **Product_mng**, é um caso curioso. É o único setor em que não houve nenhuma promoção nos últimos 5 anos, mesmo assim, possui a menor variação de avaliação e uma das maiores médias de satisfação entre os colaboradores. O setor campeão de promoções é o de **Sales**, de cada 10 promoções, 3 são para colaboradores dessa área.
##### **7.** O setor de vendas, é o que mais demitiu dentre todos na base de dados. O perfil médio de um colaborador que se desliga, possui 43% de satisfação, 3.8 anos de empresa, um decréscimo de -27.99% em relação a última avaliação e possui uma jornada de 207 horas mensais, ou 9,85h por dia.